### PR TITLE
Use unique_ptr in AttribManager.

### LINKED ISSUE #470: Memory Leak on std::move( AttribManager )
I ran into a memory leak when moving a `TriangleMesh`.
Within a plugin of mine, I do the following:
```
auto mesh = Ra::Core::Geometry::makePlaneGrid( height, width, scale );
engineMesh->loadGeometry( std::move( plane ) ); // there was already another mesh
```
This unfortunately leads to a memory leak.
After looking into the code of both `Engine::Mesh` and `Core::TriangleMesh` I found:
```
Mesh::loagGeometry( Core::TriangleMesh&& mesh ) {
m_mesh = std::move( mesh ); // which is legit
...
}
```
and
```
inline TriangleMesh& TriangleMesh::operator=( TriangleMesh&& other ) {
    m_triangles      = std::move( other.m_triangles );
    m_faces          = std::move( other.m_faces );
    m_vertexAttribs  = std::move( other.m_vertexAttribs );
    m_verticesHandle = std::move( other.m_verticesHandle );
    m_normalsHandle  = std::move( other.m_normalsHandle );
    return *this;
}
```
which is legit too!

Thus I went to `AttribManager`:
```
AttribManager& operator=( AttribManager&& m ) {
    m_attribs      = std::move( m.m_attribs ); // which is a std::vector<AttribBase*>
    m_attribsIndex = std::move( m.m_attribsIndex ); // which is a std::map<std::string, Index>
    return *this;
}
```
all seems fine there, except that we are moving raw pointers, and thus these are not deleted.

So a simple fix would be to clear `m_attribs`before moving.

A better solution would be to use `unique_ptr` instead of raw pointers for `AttribManager::value_type`

### PR DESCRIPTION
**Pull request by dlyr**
UPDATE the form below to describe your PR.

* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

Be aware that the PR request cannot be accepted if it doesn't pass the Continuous Integration tests.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Bug fix

* **What is the current behavior?** (You can also link to an open issue here)
memory leak

* **What is the new behavior (if this is a feature change)?**
use unique_ptr to have auto delete

* **Other information**:
Supersede #471, resolve #470 
To be tested

### CODE REVIEW

**Comment by nmellado** [comment]
@hoshiryu can you test this PR actually solve your memory leak please ?

File: src/Core/Utils/Attribs.hpp [hunk 1]
```diff
@@ -124,7 +124,7 @@ class Attrib : public AttribBase
     size_t getSize() override { return m_data.size(); }
 
     /// \warning Does not work for dynamic and sparse Eigen matrices.
```
Comment (hoshiryu) [inline src/Core/Utils/Attribs.hpp]: What about this warning?

Comment (dlyr) [reply src/Core/Utils/Attribs.hpp]: still true

**Comment by dlyr** [comment]
Regarding the doc comment, for a user point of view, it do not change if container use unique_ptr or raw ptr. No need to change this.

**Pull request merged**